core.py

```diff
@@ -14,6 +15,28 @@ def core_addition(input_object: str) -> str:
     
     data_dict = typeconversion.json_to_dict(input_object)
     result = calculations.addition(data_dict["number1"], data_dict["number2"])
-    output_object = typeconversion.dict_to_json(result)
+    
+    return typeconversion.dict_to_json(result)
+
 
-    return output_object
+def core_component_create_edit(input_object: str) -> str:
+    """
+    Receives a JSON object in the form defined under JSON_objects_defitnions.py for either editing a component or creating a new component
+    The answer is also a JSON object, only containing the success state, which is True or False
+    
+    :param input_object: JSON object containing all component attributes (special information to differentiate edit or create is contained in the ID, which is either -1 or the original ID
+    :type input_object: str
+    :return A JSON object containing the success state, which is True or False
+    """
+    
+    data_dict = typeconversion.json_to_dict(input_object)
+    result = {}
+    
+    if data_dict["id"] == -1:
+        result = component_handler.add_component(data_dict)
+    elif data_dict["id"] >= 0:
+        result = component_handler.edit_component(data_dict)
+    else:
+        result = {"success": False} # Optionally it could be cleaner to import the result object in the else case directly from a template
```

Comment:
Macht es Sinn, hier den error_handler in api aufzurufen?

Reply:
Unser Gedanke hier war, dass wir sonst nicht abfragen, ob evtl keine id angegeben ist (in einem falschen json format übermittelt), deswegen noch das weitere else.

Reply:
Ja genau, aber macht es Sinn statt des Hard Code  {"success": False} den error_handler in api aufzurufen? Dann hätten wir ein einheitlich definiertes Format für vorkommende Fehler.